core/App/ehsm_napi.cpp

```diff
@@ -0,0 +1,317 @@
+/*
+ * Copyright (C) 2021-2022 Intel Corporation
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ *   1. Redistributions of source code must retain the above copyright
+ *      notice, this list of conditions and the following disclaimer.
+ *   2. Redistributions in binary form must reproduce the above copyright
+ *      notice, this list of conditions and the following disclaimer in
+ *      the documentation and/or other materials provided with the
+ *      distribution.
+ *   3. Neither the name of Intel Corporation nor the names of its
+ *      contributors may be used to endorse or promote products derived
+ *      from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+#include <cstring>
+#include "base64.h"
+#include "ehsm_napi.h"
+
+#include "serialize.h"
+
+//#include "ehsm_provider.h"
+
+using namespace std;
+using namespace EHsmProvider;
+
+static char* StringTochar(std::string str)
+{
+    char *retChar = NULL;
+    if (str.size() > 0) {
+        int len = str.size();
+        retChar = (char *)malloc(len * sizeof(uint8_t));
+        if(retChar != nullptr){
+            memset(retChar, 0, len);
+            memcpy(retChar, str.c_str(), len);
+        }
+    }
+    return retChar;
+}
+
+std::string intToString(int value)
+{
+    char buf[32] = {0};
+    snprintf(buf, sizeof(buf), "%u", value);
+    std::string str = buf;
+    return str;
+}
+
+struct RetJsonObj 
```

Comment:
Do we really need to format the return value as JSON in NAPI layer? @syan10 
JS layer should format the JSON obj according to the different errors like checking the signature, time, API validation, etc.

Reply:
but how to return the error code and such above infomations? since currently the neusoft guys said it only can work with the return char*, however, we need more extra infomations need to return. so a json struct string should be ok.

Reply:
So why line@156,157 is commented out for successful return?

Reply:
just make the unit test case happy, because it need to add c++ json parser, otherwise the test cases can't get the correct prameters.

after the neusoft guys added the c++ json parser, it don't need to be commented out and then remove the StringtoChar() function.